metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -1356,11 +1370,13 @@ def _build_aggregated_measure_from_measure_source_node(
         else:
             unaggregated_measure_node = filtered_measure_source_node
 
+        # If time constraint was previously adjusted for cumulative window or grain, apply original time constraint
+        # here. Can skip if metric is being aggregated over all time.
         cumulative_metric_constrained_node: Optional[ConstrainTimeRangeNode] = None
         if (
             cumulative_metric_adjusted_time_constraint is not None
             and time_range_constraint is not None
-            and queried_linkable_specs.contains_metric_time
+            and queried_agg_time_dimension_specs
```

Comment:
Yep I'm aware! But good point that I actually I should revert the change on this line so it only applies the time constraint with `metric_time` - forgot to do that.
I had actually put up a separate issue for this since it's not directly related to the feature: https://github.com/dbt-labs/metricflow/issues/1008
Since the time constraint param isn't exposed in any of our APIs, only in the MF CLI, it seems less important to fix that issue before releasing this feature.
Note: in the integration test you noted, I used a where filter instead of a time constraint.

Reply:
I just put up a commit to raise a more appropriate error if `metric_time` is not used here.

Reply:
> Since the time constraint param isn't exposed in any of our APIs, only in the MF CLI, it seems less important to fix that issue before releasing this feature.

Right, makes sense, although we still don't want result divergence for people who do local dev with the CLI and then issue where filters in prod cloud accounts.

If I'm honest, though, I'm mostly looking ahead to predicate pushdown implementation concerns. :)

I'll take a look at the update!